### PR TITLE
feat: updated staking events to use withMetaMetrics helper

### PR DESCRIPTION
**Pull request by Matt561**
## **Description**
This PR updates the events added in #12144 to use the `withMetaMetrics` helper function.

## **Related issues**

Jira Ticket: [STAKE-876: Wrap existing Staking events with withMetaMetrics hook](https://consensyssoftware.atlassian.net/browse/STAKE-876)

## **Manual testing steps**

Prerequisite: Add `export MM_POOLED_STAKING_UI_ENABLED=true` to your local `.js.env` file.

All events can be tested by running through the stake and unstake flows.

### **Before**

N/A
### **After**

N/A
## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by nickewansmith — DISMISSED** [review]
LGTM, all found events using `withMetaMetrics` are firing.

**Review by amitabh94 — APPROVED** [review]
LGTM!

**Review by sethkfman — APPROVED** [review]
LGTM

File: app/components/UI/Stake/utils/metaMetrics/withMetaMetrics.ts [hunk 1]
```diff
@@ -27,7 +27,8 @@ const buildEvent = (e: WithMetaMetricsEvent) => {
   return eventBuilder.build();
 };
 
-export const withMetaMetrics = <T extends (...args: unknown[]) => unknown>(
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const withMetaMetrics = <T extends (...args: any[]) => any>(
```
[the remark this answers is not in the document]
Comment (Matt561) [reply app/components/UI/Stake/utils/metaMetrics/withMetaMetrics.ts:31]: I added this due to a type error when calling `withMetaMetrics` a certain way. I'll fix this in a separate PR so we can merge this 👍 

image

**Pull request merged**